libraries/AC_AutoTune/AC_AutoTune_FreqResp.h

```diff
@@ -0,0 +1,87 @@
+#pragma once
+
+/*
+ Gain and phase determination algorithm
+*/
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Math/AP_Math.h>
+
+#define AUTOTUNE_DWELL_CYCLES                6
+
+class AC_AutoTune_FreqResp {
+public:
+    // Constructor
+    AC_AutoTune_FreqResp()
+{
+    // initialize test variables
+    meas_peak_info_buffer = new ObjectBuffer<peak_info>(AUTOTUNE_DWELL_CYCLES);
+    tgt_peak_info_buffer = new ObjectBuffer<peak_info>(AUTOTUNE_DWELL_CYCLES);
+}
+
+    enum InputType {
+        DWELL = 0,                 
+        SWEEP = 1, 
+    };
+
+    // Initialize the Frequency Response Object. 
+    // Must be called before running dwell or frequency sweep tests
+    void init(InputType input_type);
+
+    // determines the gain and phase based on rate response for a dwell or sweep
+    void update_rate(float tgt_rate, float meas_rate, float tgt_freq);
+
+    // determines the gain and phase based on angle response for a dwell or sweep
+    void update_angle(float command, float tgt_angle, float meas_angle, float tgt_freq);
+
+    // enable external query if cycle is complete and freq response data are available
+    bool is_cycle_complete() { return cycle_complete;}
+
+    // reset cycle_complete flag
+    void reset_cycle_complete() { cycle_complete = false; }
+
+    // frequency response data accessors
+    float get_freq() { return curr_test_freq; }
+    float get_gain() { return curr_test_gain; }
+    float get_phase() { return curr_test_phase; }
+    float get_accel_max() { return max_accel; }
+
+private:
+    float max_target, max_meas, prev_target, prev_meas, prev_tgt_angle, prev_meas_angle;
```

Comment:
Just a formatting things but I wonder if we should perhaps separate these variable declarations onto separate lines and put comments beside each as to it's purpose.  This would be more consistent with other libraries.

Reply:
done.